src/grow_worker/worker.py

```diff
@@ -85,10 +133,15 @@ def grow_worker_task(
     #  subprocesses from being created. This does introduce the issue that if this
     #  process is killed/cancelled/revoked, the subprocess will continue as a zombie process.
     #  See https://github.com/Project-OMOTES/optimizer-worker/issues/54
-    current_process()._config['daemon'] = False  # type: ignore[attr-defined]
+    current_process()._config["daemon"] = False  # type: ignore[attr-defined]
 
     with multiprocessing.Pool(1) as pool:
-        output_esdl = pool.map(run_mesido, [input_esdl])[0]
+        try:
+            output_esdl = pool.map(run_mesido, [input_esdl])[0]
+        except SystemExit as e:
```

Comment:
This is triggered when Celery throws a SystemExit to cancel the current task. It will terminate the mesido worker subprocess with a SIGKILL to ensure it does not wait till it dies gracefully. Casadi is blocking any way to gracefully shutdown the process as it doesn't release the GIL.